metronome/checkpointing/service/src/io/iohk/metronome/checkpointing/service/CheckpointingService.scala

```diff
@@ -51,7 +57,7 @@ class CheckpointingService[F[_]: Sync: Applicative, N] private (
   override def executeBlock(
       block: Block,
       commitQC: QuorumCertificate[CheckpointingAgreement],
-      commitPath: NonEmptyList[Block.Hash]
+      commitPath: NonEmptyList[Block.Hash] //TODO: remove?
```

Comment:
I added this in case an application would need to construct a proof by retrieving block headers along the way, or something similar. Not every use case must use it, but I think it makes sense to provide the link between the `block` and the `commitQC` (which is also use case specific).